lib/Migration/Version1116Date20220110153658.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Mail\Migration;
+
+use Closure;
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version1116Date20220110153658 extends SimpleMigrationStep {
+
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 * @throws Exception
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		$table = $schema->getTable('mail_messages');
+		if (!$table->hasIndex('mail_messages_id_flags')) {
+			$table->addIndex(['mailbox_id', 'flag_important', 'flag_deleted', 'flag_seen'], 'mail_messages_id_flags');
+		}
+		if (!$table->hasIndex('mail_messages_id_flags2')) {
+			$table->addIndex(['mailbox_id', 'flag_deleted', 'flag_flagged'], 'mail_messages_id_flags2');
```

Comment:
same potential performance problems here. maybe this should all be optional indices?

Reply:
Sure thing, we could add an occ command for it for example?

Reply:
yes, albeit it would be hard for admins to ever discover those as we do not have app-provided setup checks. https://github.com/nextcloud/server/pull/25471 is probably the more integrated approach.